File: Jacobi.py
import numpy as np

def solucao(matriz):
    print("ok")

def gauss_jacobi(A, b, x0, max_iterations=100, tolerance=10e-6):
 def jacobi(a, b, x, n, tolerance):
    for k in range(n):
        next_x = [0, 0, 0]  # lista que armazena a próxima iteração
        for i in range(len(b)):
            bi = b[i]  # armazena o termo independente da linha i
            for j in range(len(b)):
                if j != i:
                    bi -= a[i][j] * x[j]  # multiplicar o valor das variáveis que não foram isoladas
            bi /= a[i][i]
            next_x[i] = bi

        # Verificar o critério de parada
        max_difference = max(abs(next_x[i] - x[i]) for i in range(len(b)))
        if max_difference < tolerance:
            print(f"Convergência alcançada na iteração {k + 1}")
            break

        x = next_x

    for i in range(len(b)):
        print(f'x_{i + 1} = {x[i]:.16f}\t')
        return next_x 

# Exemplo de uso
A = np.array([[4, 1, -1],
              [-1, 3, -1],
              [1, -1, 5]], dtype=float)
b = np.array([5, 6, 4], dtype=float)
x0 = [0, 0, 0]

#solution = gauss_jacobi(A, b, x0)
#print("Solução:", solution)

File: Jacobi.java
public class Jacobi {
    public static void main(String[] args) {
        double a[][] = {{1.0, 0.0, 0.0, 1.0}, {1.0, 1.0, 0.0, 0.0}, {0.0, 1.0, 1.0, -1.0}}; // matriz dos coeficientes
        double b[] = {35.0, 40.0, 30.0}; // matriz dos termos independentes

        double x[] = {0, 0, 0}; // iteração inicial
        int n = 100; // número máximo de iterações
        double tolerancia = 1e-6; // tolerância para o critério de parada

        jacobi(a, b, x, n, tolerancia);
    }

    public static void jacobi(double a[][], double b[], double x[], int n, double tolerance) {
        double next[] = new double[b.length]; // array que armazena a próxima iteração

        for (int k = 0; k < n; k++) {
            boolean converged = true;

            for (int i = 0; i < b.length; i++) {
                double bi = b[i]; // armazena o termo independente da linha i

                for (int j = 0; j < b.length; j++) {
                    if (j != i) {
                        bi -= a[i][j] * x[j]; // multiplicar o valor das variáveis que não foram isoladas
                    }
                }

                bi /= a[i][i];
                System.out.printf("x_%d^(%d) = %.16f\t", i + 1, k + 1, bi);
                next[i] = bi;

                // Verificar convergência
                if (Math.abs(next[i] - x[i]) >= tolerance) {
                    converged = false;
                }
            }

            System.out.println("\n");

            if (converged) {
                System.out.println("O método convergiu após " + (k + 1) + " iterações.");
                break;
            }

            for (int i = 0; i < b.length; i++) {
                x[i] = next[i];
            }
        }
    }
}